### DISCUSSION TITLE
Is Cppfront spelled with an upper case, like a name?

### DISCUSSION THREAD

**Discussion by JohelEGP** [hsutter/cppfront#848] [Q&A] [answered]
The README uses
- Cppfront 2 times,
- cppfront 13 times, and
- `cppfront` 3 times
  (when naming the executable in paths and commands,
  not to refer to the whole like Cppfront and cppfront do).

I've been torn on how to call it,
preferring to refer to the executable when possible,
since that's unambiguous.

**Comment by msadeqhe** [comment, marked as answer]
It seems *Cppfront* has uppercase *C* when it's the first word in sentences (like other sentences in English which the first letter is uppercase). Hence I guess *cppfront* seems to be the correct spelling.

**Reply by JohelEGP** [reply]
Herb, please "Mark as answer" if it's correct.

**Reply by gregmarr** [reply]
That was my initial thought when I read the question, and I confirmed, as I assume @msadeqhe did, that the only places that used `Cppfront` in the readme were the start of a sentence.

**Reply by hsutter** [reply]
Yes, that's how I've been doing it... "cfront" unless it starts a sentence.

That said, I've been wondering because I was following the example of "cfront"... the following is excessive analysis, but FWIW:
- up to the last release 3.0.3 (1994), cfront's own sources and documentation used lowercase (see: [3.0.3 installation docs](https://www.softwarepreservation.org/projects/c_plus_plus/cfront/release_3.0.3/source/INSTALLATION/view))
- but starting in D&E (also 1994) and his HOPL papers, Bjarne started capitalizing Cfront (in D&E, it's "Cfront" in 125 out of 126 occurrences, and "cfront" only in one figure on page 67, a diagram of compilation steps)
- [Wikipedia is inconsistent](https://en.wikipedia.org/wiki/Cfront) and uses both Cfront and cfront mid-sentence
- a Google search shows both, with lots of all-lower-case
- ChatGPT 4 (because why not!?) says
> The compiler name "cfront" is typically not capitalized. It's written in lowercase as "cfront." This compiler, developed by Bjarne Stroustrup at AT&T Bell Labs, was the first compiler to implement the C++ programming language. The naming convention follows a common pattern in Unix and Unix-like systems where many tools and programs are named in lowercase.
